core/src/test/scala/unit/kafka/server/epoch/util/ReplicaFetcherMockBlockingSend.scala

```diff
@@ -59,6 +65,8 @@ class ReplicaFetcherMockBlockingSend(offsets: java.util.Map[TopicPartition, Epoc
   }
 
   override def sendRequest(requestBuilder: Builder[_ <: AbstractRequest]): ClientResponse = {
+    if (!NetworkClientUtils.awaitReady(client, sourceNode, time, 500))
```

Comment:
Not clear to me why adding this is necessary?

Reply:
I made changes to `MockClient` to make its behavior closer to `NetworkClient`. One of the improvements is that you now have to call `ready` before calling `send`, which is just like `NetworkClient`.

Reply:
ack.